### PR TITLE
chore(shorebird_code_push_client): support semantically correct response codes

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

In some create/delete functions, we check for status code 200 (OK) when a 201 (created) or a 204 (no content) would be more appropriate. This PR adjusts the code push client to accept both, which will allow us to correct this server-side.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [x] 🗑️ Chore

### CODE REVIEW

**Review by erickzanardo — APPROVED** [review]
LGTM

Just wonder about unit tests that could be updated to verify for the new behaviour.

**Review by felangel — APPROVED** [review]
LGTM in addition to unit tests we should probably leave TODOs to remove the old, deprecated response codes in the future so they don't linger for a long time.

File: packages/shorebird_code_push_client/lib/src/code_push_client.dart [hunk 1]
```diff
@@ -140,7 +140,8 @@ class CodePushClient {
     final response = await _httpClient.send(request);
     final body = await response.stream.bytesToString();
 
-    if (response.statusCode != HttpStatus.ok) {
+    if (response.statusCode != HttpStatus.ok &&
```
Comment (felangel) [inline packages/shorebird_code_push_client/lib/src/code_push_client.dart]: Might also be nice to have an extension on HttpStatus like `isCreatedOrOk` to eliminate some of the duplication and then we can leave a todo on the extension method to remove it once we've rolled out the backend changes and done a force upgrade in the future.

Comment (bryanoltman) [reply packages/shorebird_code_push_client/lib/src/code_push_client.dart]: I'd be curious to hear your thoughts re: having an even more general `isSuccess` extension, something like:

```dart
bool get isSuccess => statusCode >= 200 && statusCode < 300;
```

Comment (bryanoltman) [reply packages/shorebird_code_push_client/lib/src/code_push_client.dart]: Or maybe `< 400`, depending on our feelings about redirects.

Comment (bryanoltman) [reply packages/shorebird_code_push_client/lib/src/code_push_client.dart]: Went ahead and did this. It's a quite a bit less rigid than the previous implementation but I don't think we're losing anything in terms of correctness.

**Review by erickzanardo — APPROVED** [review]

**Pull request merged**